<launch>
  <!-- Read arguments -->
  <arg name="frame" default="leddar"/>
  <arg name="range" default="50.1"/>
  <arg name="accumulations" default="6"/>
  <arg name="oversampling" default="3"/>
  <arg name="base_point_count" default="24"/>
  <arg name="threshold_offset" default="0"/>
  <arg name="acq_options" default="1"/>
  <arg name="intensity" default="7"/>

  <node name="leddar_node" pkg="leddar" type="leddar_node"
    output="screen" respawn="true">
      <!-- Set ROS parameters -->
      <param name="frame" value="$(arg frame)"/>
      <param name="range" value="$(arg range)"/>
      <param name="accumulations" value="$(arg accumulations)"/>
      <param name="oversampling" value="$(arg oversampling)"/>
      <param name="base_point_count" value="$(arg base_point_count)"/>
      <param name="threshold_offset" value="$(arg threshold_offset)"/>
      <param name="acq_options" value="$(arg acq_options)"/>
      <param name="intensity" value="$(arg intensity)"/>
  </node>
</launch>
